packages/near-membrane-base/src/blue.ts

```diff
@@ -34,56 +35,37 @@ import {
     RedConstructor,
     RedFunction,
     RedValue,
+    ProxyTargetType,
+    BlueArrayOrObject,
+    RedArrayOrObject,
 } from './types';
 
-function renameFunction(provider: RedFunction, receiver: BlueFunction) {
-    try {
-        // a revoked proxy will break the membrane when reading the function name
-        const nameDescriptor = ReflectGetOwnPropertyDescriptor(provider, 'name')!;
-        ReflectDefineProperty(receiver, 'name', nameDescriptor);
-    } catch {
-        // intentionally swallowing the error because this method is just extracting the
-        // function in a way that it should always succeed except for the cases in which
-        // the provider is a proxy that is either revoked or has some logic to prevent
-        // reading the name property descriptor.
-    }
-}
-
 const ProxyCtor = Proxy;
 const { isArray: isArrayOrNotOrThrowForRevoked } = Array;
 
-function createBlueShadowTarget(target: BlueProxyTarget): BlueShadowTarget {
+function createShadowTarget(
+    targetTypeof: string,
+    targetIsArrowFunction: boolean,
+    targetFunctionName: string | undefined,
+    targetIsArray: boolean
+): BlueShadowTarget {
     let shadowTarget;
-    if (typeof target === 'function') {
+    if (targetTypeof === 'function') {
         // this new shadow target function is never invoked just needed to anchor the realm
-        try {
-            // According the comment above, this function will never be called, therefore the
-            // code should not be instrumented for code coverage.
-            //
-            // istanbul ignore next
-            // eslint-disable-next-line func-names
-            shadowTarget = 'prototype' in target ? function () {} : () => {};
-        } catch {
-            // target is a revoked proxy
-            // According the comment above, this function will never be called, therefore the
-            // code should not be instrumented for code coverage.
-            //
-            // istanbul ignore next
-            // eslint-disable-next-line func-names
-            shadowTarget = function () {};
-        }
+        // According the comment above, this function will never be called, therefore the
+        // code should not be instrumented for code coverage.
+        //
+        // istanbul ignore next
+        // eslint-disable-next-line func-names
+        shadowTarget = targetIsArrowFunction ? () => {} : function () {};
         // This is only really needed for debugging, it helps to identify the proxy by name
-        renameFunction(target as (...args: any[]) => any, shadowTarget as (...args: any[]) => any);
+        ReflectDefineProperty(shadowTarget, 'name', {
```

Comment:
We can even guard this with a `process.env.NODE_ENV !== 'production'` guard for it (I have this in a branch) so that it is only included in dev/debug builds.

Reply:
we can probably do that in another PR once we introduce `process.env.NODE_ENV`. for now, I can just add a note here with a todo.